management/src/main/java/io/micronaut/management/endpoint/env/EnvironmentFilterSpecification.java

```diff
@@ -0,0 +1,131 @@
+        this.principal = principal;
+        this.allMasked = true;
+    }
+
+    /**
+     * @return The current {@link Principal} that is making the request (if any)
+     */
+    public @Nullable
+    Principal getPrincipal() {
+        return principal;
+    }
+
+    /**
+     * Turn on global masking. Items can be unmasked by adding their {@link Pattern} to {@link EnvironmentFilterSpecification#exclude(EnvironmentFilterNamePredicate...)}.
+     *
+     * @return itself for chaining calls.
+     */
+    public @NotNull EnvironmentFilterSpecification maskAll() {
+        allMasked = true;
+        return this;
+    }
+
+    /**
+     * Turn off global masking. Items can be masked by adding a matching {@link Predicate} to {@link EnvironmentFilterSpecification#exclude(EnvironmentFilterNamePredicate...)}.
+     *
+     * @return itself for chaining calls.
+     */
+    public @NotNull EnvironmentFilterSpecification maskNone() {
+        allMasked = false;
+        return this;
+    }
+
+    /**
+     * Adds predicates to the list of known masks.  If the {@link EnvironmentFilterSpecification#maskAll()} flag is set
+     * then this list will be used for exceptions (values in clear-text).  If the {@link EnvironmentFilterSpecification#maskNone()}
+     * flag is set then this list will be those values that are masked.
+     *
+     * @param propertySourceNamePredicates one or more predicates to match against property names for masking.
+     * @return itself for chaining calls.
+     */
+    public @NotNull EnvironmentFilterSpecification exclude(EnvironmentFilterNamePredicate... propertySourceNamePredicates) {
+        exclusions.addAll(Arrays.asList(propertySourceNamePredicates));
+        return this;
+    }
+
+    /**
+     * Adds the masking rules from legacy pre 3.3.0.  Effectively the same as calling {@link EnvironmentFilterSpecification#maskNone()}
+     * and adding predicates via {@link EnvironmentFilterSpecification#exclude(EnvironmentFilterNamePredicate...)} to mask anything containing the words
+     * {@code password, credential, certificate, key, secret} or {@code token}.
+     *
+     * @return itself for chaining calls.
+     */
+    public @NotNull EnvironmentFilterSpecification legacyMasking() {
+        allMasked = false;
+        exclusions.addAll(LEGACY_MASK_PATTERNS);
+        return this;
+    }
+
+    FilterResult test(String propertySourceName) {
+        if (exclusions.stream().anyMatch(p -> p.test(propertySourceName))) {
+            return allMasked ? FilterResult.PLAIN : FilterResult.MASK;
+        }
+        return allMasked ? FilterResult.MASK : FilterResult.PLAIN;
+    }
+
+    public static EnvironmentFilterNamePredicate regularExpressionPredicate(String pattern) {
+        return regularExpressionPredicate(pattern, 0);
+    }
+
+    public static EnvironmentFilterNamePredicate regularExpressionPredicate(String pattern, int flags) {
+        return s -> Pattern.compile(pattern, flags).matcher(s).matches();
+    }
```

Comment:
These appear now to be part of the public API? if it is the case they need javadoc

Reply:
Done.  Would you say they were necessary?  I added them to make it simpler to create the Predicates, but can see that they may add noise 🤔